chainimport/headers_import_test.go

```diff
@@ -153,3 +192,379 @@ func TestTargetStoreFreshnessDetection(t *testing.T) {
 		})
 	}
 }
+
+// TestHeaderMetadataStorage tests the header metadata storage to the file.
+// It checks that the header metadata is stored correctly to the file.
+func TestHeaderMetadataStorage(t *testing.T) {
+	t.Parallel()
+	type Prep struct {
```

Comment:
Any reason to not move have these be declared outside this function scope, and be unexported?